### DISCUSSION TITLE
Issue with GPR Metadata... I think?

### DISCUSSION THREAD

**Discussion by Lewisfa** [emanuelhuber/RGPR#45] [Q&A] [answered]
Hello!

Thanks so much for creating this! It looks so promising! I am having some issues I think with my DZT files - I am trying to do the time/depth slice interpolation. I have GPR data for a grid that is 25m (6 lines) x 15m (4 lines), scans were spaced out 5 meters apart and total 10 files.

LINES <- file.path(getwd(), "rawGPR",
                   paste0("LINE", sprintf("%02d", 1:10),".dzt"))
SU <- GPRsurvey(LINES, verbose = FALSE)
SU
SU <- GPRsurvey(LINES,verbose = FALSE)
setGridCoord(SU) <- list(xlines = 1:4,
                         xpos   = seq(0, by = 5, length.out = 4),
                         xstart = c(0, 0, 0, 0),
                         ylines = 10 + (1:4),
                         ypos   = c(0, 5, 10, 15, 20, 25),
                         ystart = c(0, 0, 0, 0, 0, 0))
plot(SU, asp = TRUE, parFid = NULL)
I received an error:
Error in `setGridCoord<-`(`*tmp*`, value = list(xlines = 1:4, xpos = c(0,  : 
  length(ylines) must be equal to length(ypos)

I think it may be because the metadata for my DTZ files is sparse, there is no information about length. Would you be able to help?

[rawGPRforgithub.zip](https://github.com/emanuelhuber/RGPR/files/6321607/rawGPRforgithub.zip)

Thanks!

**Comment by emanuelhuber** [comment, marked as answer]
Hello Lewis

Thank you for contacting me. I could solve the problem and I also had to update RGPR, so you need to re-install it. Here is a working code (you did a mistake where you set the "ylines" because `length(10 + 1:4)` is not equal 6):

```r
if(!require("devtools")) install.packages("devtools")
devtools::install_github("emanuelhuber/RGPR")

library(RGPR)

setwd("/mnt/data/RGPR/CODE/DEVELOPMENT/HELP_PEOPLE/lewisfa")


LINES <- file.path(getwd(), "rawGPR",
                   paste0("LINE", sprintf("%02d", 1:10),".DZT"))
SU <- GPRsurvey(LINES, verbose = FALSE)

# you have 10 lines
#    - 1, 2, 3, 4 = xlines -> "1:4"
#    - 5, 6, 7, 8, 9, 10 = ylines -> "5:10"
setGridCoord(SU) <- list(xlines = 1:4,
                         xpos = seq(0, by = 5, length.out = 4),
                         xstart = c(0, 0, 0, 0),
                         ylines = 5:10,  # I corrected that: 10 + (1:4),
                         ypos = c(0, 5, 10, 15, 20, 25),
                         ystart = c(0, 0, 0, 0, 0, 0))

# Now, your data do not look terrible...
plot(SU, asp = TRUE, parFid = NULL)

#-------------------------------------------------------------------
# You can also specify the length of all profiles
# Here, I assume that the trace spacing if 0.2 m and accordingly estimate
# the length of each line:
dx <- 0.2                         # trace spacing
nL <- (ntraces(SU) - 1 ) * dx     # line lengths

setGridCoord(SU) <- list(xlines = 1:4,
                         xpos = seq(0, by = 5, length.out = 4),
                         xstart = c(0, 0, 0, 0),
                         xlength = nL[1:4],
                         ylines = 5:10,  # I corrected that: 10 + (1:4),
                         ypos = c(0, 5, 10, 15, 20, 25),
                         ystart = c(0, 0, 0, 0, 0, 0),
                         ylength = nL[5:10])

plot(SU, asp = TRUE, parFid = NULL)


#=======================================================#
#     HERE IS THE SOLUTION FOR YOUR 25m x 15m GRID
#=======================================================#
setGridCoord(SU) <- list(xlines = 1:4,
                         xpos = seq(0, by = 5, length.out = 4),
                         xstart = c(0, 0, 0, 0),
                         # set the length to 25 m
                         xlength = rep(25, 4),
                         ylines = 5:10,  # I corrected that: 10 + (1:4),
                         ypos = c(0, 5, 10, 15, 20, 25),
                         ystart = c(0, 0, 0, 0, 0, 0),
                         # set the length to 15 m
                         ylength = rep(15, 6))

plot(SU, asp = TRUE, parFid = NULL)
```

Don't hesitate to contact me if you have any question or if you are missing some functionality. Any feedback helps me to improve RGPR and therefore to provide a better tool for the GPR community. Thank you.

Let me know if this issue is solved.

Best,

Emanuel

**Reply by Lewisfa** [reply]
Hi Emanuel,

Thanks so much! Ah, I feel silly, I wasn't understanding the example on the online tutorial. I am super excited to work through the rest of the data - hopefully I won't stumble again, but I really appreciate how responsive you are and your willingness to help. 

Thanks again, 
Lewis

**Reply by emanuelhuber** [reply]
Hello Lewis

No worries, that can happen to anyone... nevertheless I tried to improve the tutorial because it was not always clear. If you have any suggestions for further improvement, let me know. Thank you.
Best,
Emanuel
